File: eclipseStroreDemo/src/main/java/com/tk/learn/eclipseStroreDemo/employee/EmployeeService.java
package com.tk.learn.eclipseStroreDemo.employee;


import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EmployeeService {
    private final EmployeeRepoInt employeeRepoInt;

    public EmployeeService(EmployeeRepoInt employeeRepoInt) {
        this.employeeRepoInt = employeeRepoInt;
    }

    public List<Employee> getAllEmployees() {
        return this.employeeRepoInt.findAll();
    }


    public Employee getEmployeeById(int id) {
        return this.employeeRepoInt.findById(id);
    }

    public Employee addEmployee(Employee employee) {
        return this.employeeRepoInt.save(employee);
    }

    public Employee updateEmployee(int id,Employee employee) {
        return this.employeeRepoInt.update(id, employee);
    }

    public void deleteEmployee(int id) {
        this.employeeRepoInt.deleteById(id);
    }


}

File: eclipse-strore-demo/src/main/kotlin/com/tk/learn/demo/employee/EmployeeService.kt
package com.tk.learn.demo.employee

import org.springframework.stereotype.Service

@Service
class EmployeeService (val employeeRepo: EmployeeRepo){

    fun findAll(): List<Employee> = employeeRepo.findAll().toList()

    fun findById(id:Int) = employeeRepo.findById(id)

    fun save(employee: Employee) = employeeRepo.save(employee)


}